mivideo/mivideo_extract.py

```diff
@@ -33,32 +41,42 @@ class MiVideoExtract(object):
     '''
 
     def __init__(self):
-        udpKeyFileName: str = ENV.get('MIVIDEO', {}).get('service_account_json_filename')
-        if (udpKeyFileName is None):
-            errorMessage: str = (f'"MIVIDEO.service_account_json_filename" '
-                                 f'was not found in {CONFIG_PATH}')
-            logger.error(errorMessage)
-            raise ValueError(errorMessage)
+        self.mivideoConfig: Dict = ENV['MIVIDEO']
+        self.defaultLastTimestamp: str = self.mivideoConfig.get(
+            'default_last_timestamp', '2020-03-01T00:00:00+00:00'
+        )
+
+        dbParams: Dict = ENV['INVENTORY_DB']
+        self.appDb: DBCreator = DBCreator(dbParams)
+
+        self.kPartnerId: int
+        self.kUserId: str
+        self.kUserSecret: str
 
-        self.udpKeyFilePath: str = os.path.join(CONFIG_DIR, udpKeyFileName)
-        logger.debug(f'udpKeyFilePath: "{self.udpKeyFilePath}"')
+    def _udpConnect(self) -> bigquery.Client:
+        udpKeyFileName: str = self.mivideoConfig['udp_service_account_json_filename']
 
-        self.credentials: service_account.Credentials = (
+        udpKeyFilePath: str = os.path.join(CONFIG_DIR, udpKeyFileName)
+        logger.debug(f'udpKeyFilePath: "{udpKeyFilePath}"')
+
+        udpCredentials: service_account.Credentials = (
             service_account.Credentials.from_service_account_file(
-                self.udpKeyFilePath,
+                udpKeyFilePath,
                 scopes=['https://www.googleapis.com/auth/cloud-platform'])
         )
 
-        self.udpDb: bigquery.Client = bigquery.Client(
-            credentials=self.credentials,
-            project=self.credentials.project_id
+        udpDb: bigquery.Client = bigquery.Client(
+            credentials=udpCredentials,
+            project=udpCredentials.project_id
         )
 
-        logger.info(f'Connected to BigQuery project: "{self.udpDb.project}"')
+        logger.info(f'Connected to BigQuery project: "{udpDb.project}"')
 
-        dbParams: Dict = ENV['INVENTORY_DB']
+        return udpDb
 
-        self.appDb: DBCreator = DBCreator(dbParams)
+    def _kalturaInit(self):
+        self.kPartnerId = self.mivideoConfig['kaltura_partner_id']
+        self.kUserSecret = self.mivideoConfig['kaltura_user_secret']
```

Comment:
They only need to have values when the Kaltura API code runs.  So if that code doesn't run, I thought it would be better to leave it out.

Normally, the UDP and the Kaltura parts of this code always run whenever this MiVideo procedure is run by the main program.  However, as I was coding and testing, I would sometimes comment out one or the other in the `run()` method.  I can imagine that in the future, we might be able to tell the main program to run specific parts of a module.  (Or maybe it would be better to separate the MiVideo code into separate UDP and Kaltura modules.)

Reply:
Hm, I see. Yeah, separating at some point might make sense, but not required at this point. I also don't see a huge problem with setting them in `__init__` and then not using them.